pallets/nft/src/lib.rs

```diff
@@ -281,71 +223,145 @@ pub mod pallet {
-		MetadataTooLong,
+		/// Count of instances overflown
+		NoAvailableInstanceId,
+		/// Count of classes overflown
+		NoAvailableClassId,
+		/// Class still contains minted tokens
+		TokenClassNotEmpty,
+		/// Class does not exist
+		ClassUnknown,
+		/// Shares has to be set for liquidity mining
+		SharesNotSet,
+		/// Operation not permitted
+		NotPermitted,
 	}
 }
 
 impl<T: Config> Pallet<T> {
-	pub fn is_owner(account: &T::AccountId, token: (T::ClassId, T::TokenId)) -> bool {
-		orml_nft::Pallet::<T>::is_owner(account, token)
+	pub fn get_next_class_id() -> Result<T::NftClassId, Error<T>> {
+		NextClassId::<T>::try_mutate(|id| {
+			let current_id = *id;
+			*id = id.checked_add(&One::one()).ok_or(Error::<T>::NoAvailableClassId)?;
+			Ok(current_id)
+		})
+	}
+
+	pub fn get_next_instance_id(class_id: T::NftClassId) -> Result<T::NftInstanceId, Error<T>> {
+		NextInstanceId::<T>::try_mutate(class_id, |id| {
+			let current_id = *id;
+			*id = id.checked_add(&One::one()).ok_or(Error::<T>::NoAvailableInstanceId)?;
+			Ok(current_id)
+		})
 	}
 
-	pub fn is_locked(token: (T::ClassId, T::TokenId)) -> Result<bool, DispatchError> {
-		let token_info = orml_nft::Pallet::<T>::tokens(token.0, token.1).ok_or(Error::<T>::TokenNotFound)?;
-		Ok(token_info.data.locked)
+	pub fn class_owner(class_id: T::NftClassId) -> Option<T::AccountId> {
+		pallet_uniques::Pallet::<T>::class_owner(&class_id.into())
 	}
 
-	pub fn toggle_lock(account: &T::AccountId, token_id: (T::ClassId, T::TokenId)) -> DispatchResult {
-		let _class_info = orml_nft::Pallet::<T>::classes(token_id.0).ok_or(Error::<T>::ClassNotFound)?;
-		orml_nft::Tokens::<T>::mutate_exists(token_id.0, token_id.1, |token| -> DispatchResult {
-			if let Some(ref mut token) = token {
-				ensure!(*account == token.owner, Error::<T>::NotTokenOwner);
-				token.data.locked ^= true; // Toggle
-				Self::deposit_event(Event::TokenMintedLockToggled(
-					account.clone(),
-					token_id.0,
-					token_id.1,
-					token.data.locked,
-				));
-			}
-			Ok(())
-		})
+	pub fn instance_owner(class_id: T::NftClassId, instance_id: T::NftInstanceId) -> Option<T::AccountId> {
+		pallet_uniques::Pallet::<T>::owner(class_id.into(), instance_id.into())
+	}
+
+	pub fn do_create_class(
+		owner: T::AccountId,
+		class_type: T::ClassType,
+		metadata: BoundedVecOfUnq<T>,
+	) -> Result<(T::NftClassId, T::ClassType), DispatchError> {
+		let class_id = Self::get_next_class_id()?;
+
+		let deposit_info = match T::Permissions::has_deposit(&class_type) {
+			false => (Zero::zero(), true),
+			true => (T::ClassDeposit::get(), false),
+		};
+
+		pallet_uniques::Pallet::<T>::do_create_class(
+			class_id.into(),
+			owner.clone(),
+			owner.clone(),
+			deposit_info.0,
+			deposit_info.1,
+			pallet_uniques::Event::Created(class_id.into(), owner.clone(), owner),
+		)?;
+
+		Classes::<T>::insert(class_id, ClassInfo { class_type, metadata });
+
+		Ok((class_id, class_type))
+	}
+
+	pub fn do_mint(
```

Comment:
i would prefer to allow to specify your own id when minting NFT's, especially when uniques pallet supports this

Reply:
how about initializing the sequencer to start from e.g. 100 000 000?

Reply:
not for instance id

Reply:
you should be completely in control of your class instance id's

Reply:
i dont think that you can have both, custom id and sequencer, custom id has to be default, but you can attach sequencer of top of it if you wish

Reply:
Not my wish :) having auto ID was a strong requirement from Koda so I added it. But still not convinced custom ID is a must.